src/main/kotlin/com/dprint/services/editorservice/EditorServiceTaskQueue.kt

```diff
@@ -53,20 +62,34 @@ class EditorServiceTaskQueue(private val project: Project) {
             return
         }
         activeTasks.add(taskInfo)
-        val task =
-            object : Task.Backgroundable(project, title, true) {
-                val future = CompletableFuture<Unit>()
 
-                override fun run(indicator: ProgressIndicator) {
-                    indicator.text = title
-                    infoLogWithConsole(indicator.text, project, LOGGER)
-                    try {
-                        future.completeAsync(operation)
-                        future.get(timeout, TimeUnit.MILLISECONDS)
-                    } catch (e: TimeoutException) {
+        getBackgroundTaskQueue().run(
+            title,
+            {
+                try {
+                    operation()
+                } finally {
+                    // Always remove from active tasks, whether success or failure
+                    activeTasks.remove(taskInfo)
+                }
+            },
+            timeout,
+            true,
+        ) // canBeCancelled
+            { e ->
+                // Handle different types of exceptions
+                when (e) {
+                    is java.util.concurrent.TimeoutException -> {
                         onFailure?.invoke(e)
                         errorLogWithConsole("Dprint timeout: $title", e, project, LOGGER)
-                    } catch (e: ExecutionException) {
+                    }
+
+                    is CancellationException -> {
+                        onFailure?.invoke(e)
+                        errorLogWithConsole("Dprint cancellation: $title", e, project, LOGGER)
+                    }
+
+                    is Exception -> {
```

Comment:
We probably need to consider what kind of exceptions can occur for the coroutine implementation.